Python/fileutils.c

```diff
@@ -2390,13 +2390,18 @@ _Py_find_basename(const wchar_t *filename)
    to be the end of the path. 'normsize' will be set to contain the
    length of the resulting normalized path. */
 wchar_t *
-_Py_normpath_and_size(wchar_t *path, Py_ssize_t size, Py_ssize_t *normsize)
+_Py_normpath_and_size(wchar_t *path, Py_ssize_t size, Py_ssize_t start, Py_ssize_t *normsize)
 {
     assert(path != NULL);
```

Comment:
```suggestion
    assert(path != NULL);
    assert(start>=0);
```
Just to be sure *I checked the code and if `start<0` then currently nothing happens which is ok)

Reply:
Is this supposed to raise an error when I call `_path_abspath("", -1)`?

Reply:
The internal `_Py_normpath_and_size()` function should use an `assert`, but builtin `_path_abspath()` should raise `ValueError` if `start` is negative.

Reply:
If you plan to to continue exposing `start`, as just an extended version of `_path_normpath()`, the function name should be something like `path_normpath_with_start()`. I proposed the name ``_path_abspath()` when I thought the plan was to move all of the implementation into C, including getting and joining the working directory, as is implemented by lower-level `_Py_abspath()` in "Python/fileutils.c".

Reply:
That was the plan, I simply didn't know how to do that.